code/_helpers/type2type.dm

```diff
@@ -215,3 +216,38 @@
 			else
 				return /datum
 	return text2path(copytext(string_type, 1, last_slash))
+
+///Lets us safley attempt to convert an atom to another type. This basically deletes that atom and replaces it with an atom of the new type. Will transfer mind if needed. convertatom2type(atom/target_atom, type, convert_message)
+/proc/convertatom2type(atom/target_atom, type, convert_message, no_transfer = FALSE, force_mind = FALSE)
+	if(!istype(target_atom))
+		return FALSE
+	if(!ispath(type))
+		return FALSE
+
+	var/atom_loc = get_turf(target_atom)
+	var/atom/new_atom = new type(atom_loc)
+
+	if(ismob(target_atom))
+		var/mob/target_mob = target_atom
+		var/mob/new_mob = new_atom
+
+		for(var/obj/item/W in target_mob)
+			target_mob.drop_from_inventory(W)
+		target_mob.regenerate_icons()
+
+		if(!target_mob.mind)
+			qdel(target_atom)
+			return new_atom
+		if(convert_message)
+			to_chat(target_mob, convert_message)
+		if(!istype(new_mob) || (!new_mob.mind && !force_mind))
+			target_mob.ghostize()
+			qdel(target_atom)
+			return new_atom
```

Comment:
Why regenerate icons on mob that's going to be deleted anyways and also why use target_atom for qdel if you already have target_mob local var?